dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/TrackerExportController.java

```diff
@@ -0,0 +1,101 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import static org.hisp.dhis.webapi.controller.tracker.TrackerControllerSupport.RESOURCE_PATH;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.hisp.dhis.common.Pager;
+import org.hisp.dhis.dxf2.events.trackedentity.TrackedEntityInstanceService;
+import org.hisp.dhis.trackedentity.TrackedEntityInstanceQueryParams;
+import org.hisp.dhis.tracker.domain.TrackedEntity;
+import org.hisp.dhis.tracker.domain.mapper.TrackedEntityMapper;
+import org.hisp.dhis.tracker.domain.web.PagingWrapper;
+import org.hisp.dhis.webapi.controller.event.TrackedEntityInstanceCriteria;
+import org.hisp.dhis.webapi.controller.event.mapper.TrackedEntityCriteriaMapper;
+import org.hisp.dhis.webapi.service.ContextService;
+import org.hisp.dhis.webapi.service.TrackedEntityInstanceSupportService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequestMapping( value = RESOURCE_PATH )
+@RequiredArgsConstructor
+public class TrackerExportController
+{
+    private final ContextService contextService;
+
+    private final TrackedEntityCriteriaMapper criteriaMapper;
+
+    private final TrackedEntityInstanceService trackedEntityInstanceService;
+
+    private final TrackedEntityMapper trackedEntityMapper;
+
+    private final TrackedEntityInstanceSupportService trackedEntityInstanceSupportService;
+
+    @GetMapping( produces = APPLICATION_JSON_VALUE )
+    PagingWrapper<TrackedEntity> getInstances( TrackedEntityInstanceCriteria criteria )
```

Comment:
No `public`?

Reply:
it's not necessary to have public visibility for a controller method since Spring will be able to invoke it anyway. Ideally lower visibility is better. Also since test is in same package it will be visible from testing classes. If there's a valid point to make it public I can change it.